src/main/java/filter/expression/Qualifier.java

```diff
@@ -214,16 +192,16 @@ public static void setCurrentTime(LocalDateTime dateTime) {
     }
 
     public boolean isEmptyQualifier() {
-        return name.isEmpty() && content.isPresent() && content.get().isEmpty();
+        return type == QualifierType.EMPTY;
     }
 
     public boolean isFalseQualifier() {
-        return name.equalsIgnoreCase("false") && content.isPresent() && content.get().isEmpty();
+        return type == QualifierType.FALSE;
```

Comment:
Okay, sure then. We can do that.
On Dec 23, 2015 1:20 AM, "Darius Foo" <notifications@github.com> wrote:

> In src/main/java/filter/expression/Qualifier.java
> <https://github.com/HubTurbo/HubTurbo/pull/1159#discussion_r48275728>:
>
> >      }
> >
> >      public boolean isFalseQualifier() {
> > -        return name.equalsIgnoreCase("false") && content.isPresent() && content.get().isEmpty();
> > +        return type == QualifierType.FALSE;
>
> Okay, yeah, having to having to construct them manually would result in
> more boilerplate.
>
> I'm still considering changing == EMPTY to isEmptyQualifier. Though we
> should use Qualifier.EMPTY whenever possible, new
> Qualifier(QualifierType.EMPTY, "") is an equally valid empty qualifier,
> and == would not allow it. Just seems a bit less brittle?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/HubTurbo/HubTurbo/pull/1159/files#r48275728>.
>